src/domain/entities/Questionnaire/QuestionnaireRules.ts

```diff
@@ -0,0 +1,269 @@
+
+    const leftOperand = condition
+        .substring(0, condition.indexOf(operator))
+        .replaceAll("'", "")
+        .trim();
+
+    const rightOperandStr = condition
+        .substring(condition.indexOf(operator))
+        .replace(operator, "")
+        .replaceAll("'", "")
+        .trim();
+
+    // Handle right operands boolean values of "1" and "0" for true and false
+    const rightOperand =
+        leftOperand === "true" && rightOperandStr === "1"
+            ? "true"
+            : leftOperand === "false" && rightOperandStr === "0"
+            ? "false"
+            : rightOperandStr;
+
+    switch (operator) {
+        case "!=": {
+            return leftOperand !== rightOperand;
+        }
+        case "==": {
+            return leftOperand === rightOperand;
+        }
+        case ">": {
+            try {
+                return parseFloat(leftOperand) > parseFloat(rightOperand);
+            } catch {
+                return leftOperand > rightOperand;
+            }
+        }
+        case ">=": {
+            try {
+                return parseFloat(leftOperand) >= parseFloat(rightOperand);
+            } catch {
+                return leftOperand >= rightOperand;
+            }
+        }
+        case "<": {
+            try {
+                return parseFloat(leftOperand) < parseFloat(rightOperand);
+            } catch {
+                return leftOperand < rightOperand;
+            }
+        }
+        case "<=": {
+            try {
+                return parseFloat(leftOperand) <= parseFloat(rightOperand);
+            } catch {
+                return leftOperand <= rightOperand;
+            }
+        }
+        default:
+            throw new Error(`Operator ${operator} not handled`);
+    }
+};
+
+export const parseCondition = (
+    condition: string,
+    updatedQuestion: Question,
+    questions: Question[]
+): boolean => {
+    // Create a regular expression from D2_FUNCTIONS array
+    const d2FunctionsRegex = new RegExp(`(?<!${D2_FUNCTIONS.join("|")})\\(`);
+
+    // Handle parentheses as long as they are not immediately preceded by a value in D2_FUNCTIONS array
+    while (condition.search(d2FunctionsRegex) !== -1) {
+        condition = condition.replace(
+            new RegExp(`(?<!${D2_FUNCTIONS.join("|")})\\(([^()]+)\\)`, "g"),
+            (_, subCondition) => {
+                return parseCondition(subCondition, updatedQuestion, questions) ? "true" : "false";
+            }
+        );
+    }
+
+    // Split condition into sub-conditions based on logical operators
+    const andConditions = condition.split("&&").map(subCondition1 => {
+        const orConditions = subCondition1.split("||").map(subCondition2 => {
+            const notCondition = subCondition2.trim().startsWith("!");
+            if (notCondition) {
+                subCondition2 = subCondition2.trim().substring(1);
+            }
+
+            let result: boolean;
```

Comment:
When doing this, typically you want a function: `const result = someFunction(...)`